app/services/tax_rate_updater.rb

```diff
@@ -20,7 +20,9 @@ def updated_rate
   end
 
   def transition_rate!
-    updated_rate.save && current_rate.destroy
+    ActiveRecord::Base.transaction do
+      updated_rate.save && current_rate.destroy
+    end
```

Comment:
It was just a suggestion. Without transactions there can't deadlocks :joy:

Reply:
I think we're safe on the deadlocks front here, no? The two objects don't really depend on each other, the first one hasn't even been saved yet at this point, and the second one should not be simultaneously updated by some other process. Eg; we don't have lots of other processes that might be modifying a TaxRate record at the same time.

Reply:
yeah, should be fine but... you never know :shrug: